src/contrib/cluster/Akka.Cluster.Sharding.Tests.MultiNode/ShardedDaemonProcessSpec.cs

```diff
@@ -0,0 +1,174 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="ClusterShardingSpec.cs" company="Akka.NET Project">
+//     Copyright (C) 2009-2020 Lightbend Inc. <http://www.lightbend.com>
+//     Copyright (C) 2013-2020 .NET Foundation <https://github.com/akkadotnet/akka.net>
+// </copyright>
+//-----------------------------------------------------------------------
+
+using System;
+using System.Linq;
+using Akka.Actor;
+using Akka.Cluster.TestKit;
+using Akka.Cluster.Tools.Singleton;
+using Akka.Configuration;
+using Akka.Remote.TestKit;
+using FluentAssertions;
+
+namespace Akka.Cluster.Sharding.Tests.MultiNode
+{
+    public class ShardedDaemonProcessSpecConfig : MultiNodeConfig
+    {
+        public RoleName First { get; }
+        public RoleName Second { get; }
+        public RoleName Third { get; }
+
+        public ShardedDaemonProcessSpecConfig()
+        {
+            First = Role("first");
+            Second = Role("second");
+            Third = Role("third");
+
+            CommonConfig = DebugConfig(false)
+                .WithFallback(ConfigurationFactory.ParseString(@"
+                    akka.loglevel = INFO
+                    akka.cluster.sharded-daemon-process {{
+                      sharding {{
+                        # First is likely to be ignored as shard coordinator not ready
+                        retry-interval = 0.2s
+                      }}
+                      # quick ping to make test swift
+                      keep-alive-interval = 1s
+                    }}
+                "))
+                .WithFallback(ClusterSharding.DefaultConfig())
+                .WithFallback(ClusterSingletonManager.DefaultConfig())
+                .WithFallback(MultiNodeClusterSpec.ClusterConfig());
+        }
+    }
+
+    public class ShardedDaemonProcessMultiNode : ShardedDaemonProcessSpec
+    {
+        public ShardedDaemonProcessMultiNode() : this(new ShardedDaemonProcessSpecConfig()) { }
+        protected ShardedDaemonProcessMultiNode(ShardedDaemonProcessSpecConfig config) : base(config, typeof(ShardedDaemonProcessMultiNode)) { }
+    }
+
+    public abstract class ShardedDaemonProcessSpec : MultiNodeClusterSpec
+    {
+        private readonly ShardedDaemonProcessSpecConfig _config;
+
+        protected ShardedDaemonProcessSpec(ShardedDaemonProcessSpecConfig config, Type type)
+            : base(config, type)
+        {
+            _config = config;
+        }
+
+        [MultiNodeFact]
+        public void ShardedDaemonProcess_Specs()
+        {
+            ShardedDaemonProcess_Should_Init_Actor_Set();
+        }
+
+        public void ShardedDaemonProcess_Should_Init_Actor_Set()
+        {
+            // HACK
+            RunOn(() => FormCluster(_config.First, _config.Second, _config.Third), _config.First);
```

Comment:
Test does not pass unless I put `FormCluster()` inside a `RunOn()`. I'm missing something.

Reply:
I'll take a look during review

Reply:
Not sure, off the top of my head, what the issue might be here - what type of error do you get?

Reply:
I only get one `ProcessActorEvent` instead of the 4 expected (one per `ProcessActor`), so the test fails with timeout.